packages/remix-dev/cli/migrate/migrations/convert-to-javascript/transform/createVariableDeclarationIdentifier.ts

```diff
@@ -24,7 +27,9 @@ export const createVariableDeclarationIdentifier = (
               : []
           )[0].local?.name || ""
       ),
-      callExpression
+      isDefaultImport
+        ? j.memberExpression(callExpression, j.identifier("default"))
+        : callExpression
```

Comment:
```ts
import Foo from 'foo';
import * as Bar from 'bar';

// =>

const Foo = require('foo').default;
const Bar = require('bar');
```

Reply:
This won't always be correct 😬 I think we've made a big mistake by building this into our CLI. There are too many edge cases 😬

Reply:
> This won't always be correct 😬

Can you give some examples where this won't be correct?
I'll try to add them.

> I think we've made a big mistake by building this into our CLI.

We wanted to support JS & TS & (as you know) having this migration was the cleanest way of converting all files in a template/stack to JS.
Otherwise each stack had to do the conversion on its own.

I think the decision for doing it in our CLI was a good one, but it's a pain to think about every possible thing. 🙈

> There are too many edge cases 😬

If only there was a package/library that could convert to ES5 without making the file a mess 🤔😅

Reply:
I do think that the TS->JS conversion is better suited as a standalone tool, in its own repo, provided as-is without covering all edge cases, rather than being included in our migrations. I think "migrations" concept in our CLI should be reserved for helping automate upgrading through breaking changes.

Reply:
That said, we currently offer the `--no-typescript` option during create... so we'd have to deprecate / remove that option to get out of the business of doing the TS->JS conversion.

Reply:
We supported both JS & TS version of our templates/stacks from v1.0.0, so that's why I added the new `convert-to-javascript` migration.

Now that we come across these extra changes that need to be made, I can see a separate package working for stacks.
We would still have the "problem" that our templates will need to use it as well.

We could definitely stop officially supporting JS, but that would indeed be a breaking change, so needs to wait for v2.

If we want, we can already create a separate package that's solely responsible for the JS conversion (or we could put it in a `scripts` package if that's preferred) & let the CLI use this new package.

Reply:
I do like the idea of moving this all to a separate package so it can be quickly iterated upon and remix can depend on that package until v2 when it can be removed.

Reply:
A separate Remix package or like a separate personal package?

Reply:
Probably a separate Remix package I think. But anyway, I think we should probably just merge this and move forward.